jmclient/jmclient/wallet.py

```diff
@@ -255,19 +255,41 @@ def read_wallet_file_data(self, filename, pwd=None, wallet_dir=None):
                 self.index_cache += [[0,0]] * (
                     self.max_mix_depth - len(self.index_cache))
         password_key = btc.bin_dbl_sha256(pwd)
-        encrypted_seed = walletdata['encrypted_seed']
+        if 'encrypted_seed' in walletdata: #accept old field name
+            encrypted_entropy = walletdata['encrypted_seed']
+        elif 'encrypted_entropy' in walletdata:
+            encrypted_entropy = walletdata['encrypted_entropy']
         try:
-            decrypted_seed = decryptData(
+            decrypted_entropy = decryptData(
                     password_key,
-                    encrypted_seed.decode('hex')).encode('hex')
+                    encrypted_entropy.decode('hex')).encode('hex')
             # there is a small probability of getting a valid PKCS7
             # padding by chance from a wrong password; sanity check the
             # seed length
-            if len(decrypted_seed) != 32:
+            if len(decrypted_entropy) != 32:
                 raise ValueError
         except ValueError:
             log.info('Incorrect password')
             return None
+
+        if 'encrypted_mnemonic_extension' in walletdata:
+            try:
+                cleartext = decryptData(password_key,
+                    walletdata['encrypted_mnemonic_extension'].decode('hex'))
+                #theres a small chance of not getting a ValueError from the wrong
+                # password so also check the sum
+                if cleartext[-9] != '\xff':
+                    raise ValueError
+                chunks = cleartext.split('\xff')
+                if len(chunks) < 2 or cleartext[-8:] != btc.dbl_sha256(chunks[1])[:8]:
```

Comment:
Won't `len(chunks)` always be 3? If padding is '' then `cleartext.split('\xff')` will still return 3 elements in the array: `['', me, checksum]`

Reply:
Not if some gibberish is returned by decryptData()